src/browser/wpe/opencdm/open_cdm.cpp

```diff
@@ -163,12 +163,9 @@ int OpenCdm::Load(std::string& responseMsg) {
   return ret;
 }
 
-int OpenCdm::Update(unsigned char* pbResponse, int cbResponse, std::string& responseMsg)
+OpenCdm::KeyStatus OpenCdm::Update(unsigned char* pbResponse, int cbResponse, std::string& responseMsg)
```

Comment:
This needs to be a vector of key statuses because there can be more than one key per session (and there is with YT2018 tests at least). The main architectural problem I found was the prefixed EME messages, it's very ugly to use this method for communicating structured data unfortunately.

Reply:
I think this could be done when, as you say in the general comment, this supports more than one key per session.